src/drivers/dw/dma.c

```diff
@@ -1073,7 +1076,7 @@ static int dw_dma_get_data_size(struct dma_chan_data *channel,
 	tr_dbg(&dwdma_tr, "dw_dma_get_data_size(): dma %d channel %d get data size",
 	       channel->dma->plat_data.id, channel->index);
 
-	irq_local_disable(flags);
+	spin_lock_irq(&channel->dma->lock, flags);
```

Comment:
are we allowed to take a spinlock while writing a trace in dw_dma_avail_data_size() and dw_dma_free_data_size()

Reply:
DMA trace will go asynchronously and etrace works without any internal spinlocks AFAIK. I don't think this is an issue.

Reply:
@plbossart this is a per-channel spin-lock, and as far as I understand DW DMA isn't used for writing traces? So there should be no recursion?

Reply:
DW DMA is used for trace, but trace DMA work is scheduled so should not deadlock here.

Reply:
ok then.